TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/workflow/services/workflow_service.py

```diff
@@ -0,0 +1,860 @@
+            workflow_name = result.get("metadata", {}).get("name", "unknown")
+            workflow_uid = result.get("metadata", {}).get("uid", "unknown")
+
+            logger.info(f"Workflow {workflow_name} submitted successfully")
+
+            return WorkflowSubmitResult(
+                success=True,
+                workflow_name=workflow_name,
+                workflow_uid=workflow_uid,
+                namespace=namespace,
+                phase=None,
+                output_message=None,
+                error=None
+            )
+
+        except requests.exceptions.RequestException as e:
+            error_msg = f"Failed to submit workflow: {e}"
+            logger.error(error_msg)
+
+            if hasattr(e, 'response') and e.response is not None:
+                try:
+                    error_detail = e.response.json()
+                    error_msg = f"Failed to submit workflow: {error_detail}"
+                except Exception:
+                    error_msg = f"Failed to submit workflow: {e.response.text}"
+
+            return WorkflowSubmitResult(
+                success=False,
+                workflow_name="",
+                workflow_uid="",
+                namespace=namespace,
+                phase=None,
+                output_message=None,
+                error=error_msg
+            )
+
+        except Exception as e:
+            error_msg = f"Unexpected error submitting workflow: {e}"
+            logger.exception(error_msg)
+
+            return WorkflowSubmitResult(
+                success=False,
+                workflow_name="",
+                workflow_uid="",
+                namespace=namespace,
+                phase=None,
+                output_message=None,
+                error=error_msg
+            )
+
+    def list_workflows(
+        self,
+        namespace: str,
+        argo_server: str,
+        token: Optional[str] = None,
+        insecure: bool = False,
+        exclude_completed: bool = False,
+        phase_filter: Optional[str] = None
+    ) -> WorkflowListResult:
+        """List workflows in a namespace via Argo Workflows REST API.
+
+        Args:
+            namespace: Kubernetes namespace
+            argo_server: Argo Server URL
+            token: Bearer token for authentication
+            insecure: Whether to skip TLS verification
+            exclude_completed: Only return running workflows
+            phase_filter: Filter by specific phase
+
+        Returns:
+            WorkflowListResult dict with success status, workflows list, count, and error
+        """
+        try:
+            headers = self._prepare_headers(token)
+            url = f"{argo_server}/api/v1/workflows/{namespace}"
+
+            logger.info(f"Listing workflows in namespace {namespace} (exclude_completed={exclude_completed})")
+            logger.debug(f"List request URL: {url}")
+
+            response = requests.get(url, headers=headers, verify=not insecure)
```

Comment:
I'm not sure if argo can do the same sort of filtering on metadata that k8s does.  Other than consistency, is there any reason that you would need to continue using argo directly to get workflows?  I suspect that it might be less impactful to do argo.  Just for my own knowledge.